File: Domain/Entities/Comment.cs
using System.ComponentModel.DataAnnotations.Schema;
using Domain.Aggregates;
using Domain.Models;

namespace Domain.Entities
{
    [Table("Comments")]
    public class Comment : Entity<int>
    {
        public int UserId { get; set; }
        [ForeignKey("UserId")]
        public virtual User? User { get; set; }
        public string? Description { get; set; }
        public DateTime CreatedDate { get; set; }
        public int Likes { get; set; }
        public int PostId { get; set; }
        [ForeignKey("PostId")]
        public virtual Post? Post { get; set; }
    }
}

File: UI/src/app/core/models/Comment.ts
export interface Comment {
  id: number;
  userId: number;
  description: string | null;
  createdDate: string;
  likesCount: number;
  postId: number;
}

export interface CommentCreate {
  description: string | null;
  postId: number;
}

export interface CommentUpdate {
  description: string;
}
